### PR TITLE
docs(gcb): Add artifacts to GCB docs

### PR DESCRIPTION
**Pull request by ezimanyi**
@ttomsu : You'd suggested improving the documentation here so it's more clear how to set up artifact support.  I just ran through the full set-up myself and everything worked following these steps.

### CODE REVIEW

**Review by ttomsu — APPROVED** [review]

File: setup/ci/gcb.md [hunk 1]
```diff
@@ -66,6 +66,9 @@ on the status of your build.
...
 7. In the **Payload Constraints** field, you can enter any of the top-level fields from the
 [Build object documentation](https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.builds#resource-build)
-as the key, and a Java regular expression as the value. For example, to match on an image produced,
-you can use the key `images` and the value `gcr\.io/my-project-id/my-application:latest` (replacing
-`my-project-id` and `my-application` with appropriate values).
+as the key, and a Java regular expression as the value.
+
+8. In the **Expected Artifacts** field, you can add any build artifacts as expected artifacts. For example,
+if the build produces a docker image, you can add an expected artifact of type *Docker* with a value of
+`gcr.io/my-project-id/my-application` (replacing `my-project-id` and `my-application` with appropriate values). You
```
Comment (lwander) [inline setup/ci/gcb.md]: I would call out that images can be referenced by `:tag` here too -- we submit artifacts for both `gcr.io/image` and `gcr.io/image:tag` with references to `gcr.io/image@sha256:abc...`

File: setup/ci/gcb.md [hunk 2]
```diff
@@ -66,6 +66,10 @@ on the status of your build.
 
 7. In the **Payload Constraints** field, you can enter any of the top-level fields from the
 [Build object documentation](https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.builds#resource-build)
-as the key, and a Java regular expression as the value. For example, to match on an image produced,
-you can use the key `images` and the value `gcr\.io/my-project-id/my-application:latest` (replacing
-`my-project-id` and `my-application` with appropriate values).
+as the key, and a Java regular expression as the value.
+
+8. In the **Expected Artifacts** field, you can add any build artifacts as expected artifacts. For example,
+if the build produces a docker image, you can add an expected artifact of type *Docker* with a value of
```
Comment (dorbin) [inline setup/ci/gcb.md]: s/docker/Docker/

File: setup/ci/gcb.md [hunk 3]
```diff
@@ -66,6 +66,10 @@ on the status of your build.
...
+`gcr.io/my-project-id/my-application:tag` (replacing `my-project-id`, `my-application`, and `tag` with
+appropriate values). You will then be able to [use the produced image](/reference/artifacts/in-pipelines/)
```
Comment (dorbin) [inline setup/ci/gcb.md]: s/will then be able to/can then/

**Review by dorbin — APPROVED** [review]
Couple of nits, but LGTM

**Pull request merged**